### PR TITLE
(fix) O3-2937: Make clinical status mandatory in the conditions form

### PR DESCRIPTION
**Pull request by denniskigen**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary

`Clinical status` should be a mandatory field in the conditions form. The FHIR [Conditions resource terminology bindings](https://www.hl7.org/fhir/condition.html#tx) mark it as a required field. It also doesn't make much sense to record a condition without recording the status of the condition.

This PR makes that possible by adding the necessary validation to the form so that an error gets shown when the form is submitted without providing a condition name and a clinical status.

## Screenshots

https://github.com/openmrs/openmrs-esm-patient-chart/assets/8509731/42b6ceb1-2903-4078-85da-8cf66bce1090

## Related Issue

https://openmrs.atlassian.net/browse/O3-2937

## Other

From the [FHIR Condition resource elements](https://www.hl7.org/fhir/condition-definitions.html#9.2.7), `Verification status` is another required field:

CleanShot 2024-03-25 at 12  49 44@2x

Should we add this to the Conditions form?

### CODE REVIEW

File: packages/esm-patient-conditions-app/src/conditions/conditions-widget.component.tsx [hunk 1]
```diff
@@ -363,11 +367,11 @@ const ConditionsWidget: React.FC<ConditionsWidgetProps> = ({
...
 };
 
-function RequiredLabel({ isRequired, text, t }: RequiredLabelProps) {
+function RequiredFieldLabel({ editing, label, t }) {
   return (
     <>
-      <span>{text}</span>
-      {isRequired && (
+      <span>{label}</span>
+      {!editing && (
```
Comment (ibacher) [inline packages/esm-patient-conditions-app/src/conditions/conditions-widget.component.tsx:374]: I'm not sure I quite understand the reason for changing things based on whether someone is editing or not? It seems like a required field is required regardless and should be marked that way no?

**Review by ibacher — APPROVED** [review]
LGTM

File: packages/esm-patient-conditions-app/src/conditions/conditions-widget.component.tsx (hunk 1 above)
Comment (denniskigen) [reply packages/esm-patient-conditions-app/src/conditions/conditions-widget.component.tsx:374]: That's right, @ibacher. A bad rebase was the culprit here.

**Comment by denniskigen** [comment]
Thanks, @ibacher!

**Pull request merged**